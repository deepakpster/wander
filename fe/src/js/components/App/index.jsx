import React from 'react';
import PropTypes from 'prop-types';

export default class App extends React.Component {
  componentDidMount(){
    this.props.dashboardActions.fetchCovidData()
  }
  render() {
    const {statistics} = this.props.dashboard;
    return (<div className={`container`}>
      <nav>
        <button className="btn btn-secondary" onClick={this.props.authActions.logout}>Logout</button>
      </nav>
      <div className="row">
        {
          statistics.map((s,sIdx)=>{
            return <div key={`c-${sIdx}`} className="col-md-6 card">
                <div className="card-header">{s.state_name || 'India (All States)'}</div>
                <ul class="list-group list-group-flush">
                <li class="list-group-item">Active(Total):{s.active}</li>
                  <li class="list-group-item">Active(New):{s.new_active}</li>
                  <li class="list-group-item">Positive(Total):{s.positive}</li>
                  <li class="list-group-item">Positive(New):{s.new_positive}</li>
                  <li class="list-group-item">Cured(Total):{s.cured}</li>
                  <li class="list-group-item">Cured(New):{s.new_cured}</li>
                  <li class="list-group-item">Death(Total):{s.death}</li>
                  <li class="list-group-item">Death(New):{s.new_death}</li>
                </ul>
              </div>
          })
        }
      </div>
      </div>
    );
  }
}
