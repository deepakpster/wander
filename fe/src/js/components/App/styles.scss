:local {
    .body{
        overflow: auto;
        max-height: calc(100vh - 55px);
    }
}
