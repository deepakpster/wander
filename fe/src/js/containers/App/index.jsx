import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import App from '../../components/App';
import { withRouter } from 'react-router-dom';
import { authActions, dashboardActions } from '../../actions';


export default withRouter(connect(
  (state, props) => ({
      ...state,
      ...props,
  }),
  (dispatch => ({
      dashboardActions: bindActionCreators(dashboardActions, dispatch),
      authActions: bindActionCreators(authActions, dispatch)
  })),
)(App));
