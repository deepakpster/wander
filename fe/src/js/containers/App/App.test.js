import React from 'react';
import { shallow } from 'enzyme';
import App from './../App';

describe('App', () => {
  it("Should render without throwing an error", () => {
    expect(shallow(
      <App/>
    )).not.toBeNull();
  });

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App debug />);
  
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with no props', () => {
    const component = shallow(<App/>);
    
    expect(component).toMatchSnapshot();
  });
});
