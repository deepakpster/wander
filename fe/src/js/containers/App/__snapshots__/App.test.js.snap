// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`App should render correctly with no props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
